vm/src/obj/objdict.rs

```diff
@@ -295,29 +215,43 @@ impl PyDictRef {
             }
         }
     }
+
+    /// Take a python dictionary and convert it to attributes.
+    pub fn to_attributes(self) -> PyAttributes {
+        let mut attrs = PyAttributes::new();
+        for (key, value) in self.get_key_value_pairs() {
+            let key = objstr::get_value(&key);
+            attrs.insert(key, value);
+        }
+        attrs
+    }
 }
 
 impl DictProtocol for PyDictRef {
     fn contains_key(&self, k: &str) -> bool {
-        content_contains_key_str(&self.entries.borrow(), k)
+        self.entries.borrow().get(k).is_some()
     }
 
     fn get_item(&self, k: &str) -> Option<PyObjectRef> {
-        content_get_key_str(&self.entries.borrow(), k)
+        match self.entries.borrow().get(k) {
+            Some(v) => Some(v.1.clone()),
+            None => None,
+        }
     }
 
     fn get_key_value_pairs(&self) -> Vec<(PyObjectRef, PyObjectRef)> {
         get_key_value_pairs(self.as_object())
     }
 
     // Item set/get:
-    fn set_item(&self, ctx: &PyContext, key: &str, v: PyObjectRef) {
-        let key = ctx.new_str(key.to_string());
-        set_item_in_content(&mut self.entries.borrow_mut(), &key, &v);
+    fn set_item(&self, ctx: &PyContext, key_str: &str, v: PyObjectRef) {
```

Comment:
Note that in order to set an item, we in future probably require a vm reference, not a context.

Reply:
That's the next step. (Now pushed.)